lib/openrtb2_3/bid.js

```diff
@@ -41,7 +42,30 @@ Bid.prototype.replaceMacros = function(string, valuesMap){
   return replaceMacros(string, valuesMap);
 };
 
-var BidBuilder = function(){};
+Bid.prototype.isBelowFloor = function(bidfloor) {
+  return this.price < bidfloor;
+};
+
+Bid.prototype.hasAdm = function() {
+  // Double ! is intentional
+  return !!this.adm && typeof this.adm === 'string';
+};
+
+Bid.prototype.isBlockedAdomain = function(badv) {
+  if (!Array.isArray(badv)) {
+    throw new Error('badv should be an array');
+  }
+  return badv.some(a => ~this.adomain.indexOf(a));
+};
+
+Bid.prototype.isBlockedCat = function(bcat) {
+  if (!Array.isArray(bcat)) {
+    throw new Error('bcat should be an array');
+  }
+  return bcat.some(c => ~this.cat.indexOf(c));
+};
+
+const BidBuilder = function(){};
```

Comment:
I like it too:) It feels right. I wanted to move the native asset validation here too but I don't want to deviate from priorities for now.